Src/SwqlStudio/ConnectionInfo.cs

```diff
@@ -233,10 +233,11 @@ public static void DoWithExceptionTranslation(Action action)
                                            });
         }
 
-        public static T DoWithExceptionTranslation<T>(Func<T> action)
+        public static T DoWithExceptionTranslation<T>(Func<T> action, bool retryOnConnectionError = true)
         {
             string msg;
             Exception inner;
+            bool couldRetry = false;
```

Comment:
this is just me being nitpicky, but I'd say 'couldRetry' should be the parameter, and local variable should be 'retryOnConnectionError'.

Reply:
I disagree. The parameter from the caller indicates whether a retry is _wanted_. The local variable indicates whether the current failure makes sense to retry. We will repeat the operation if _both_ conditions hold.